HDDS-1735. Create separate unit and integration test executor dev-support script

elek opened the pull request:
hadoop-ozone/dev-support/checks directory contains multiple helper script to execute different type of testing (findbugs, rat, unit, build).

They easily define how tests should be executed, with the following contract:

 * The problems should be printed out to the console

 * in case of test failure a non zero exit code should be used

 

The tests are working well (in fact I have some experiments with executing these scripts on k8s and argo where all the shell scripts are executed parallel) but we need some update:

 1. Most important: the unit tests and integration tests can be separated. Integration tests are more flaky and it's better to have a way to run only the normal unit tests

 2. As HDDS-1115 introduced a pom.ozone.xml it's better to use them instead of the magical "am pl hadoop-ozone-dist" trick--

 3. To make it possible to run blockade test in containers we should use - T flag with docker-compose

 4. checkstyle violations are printed out to the console

See: https://issues.apache.org/jira/browse/HDDS-1735

hadoop-yetus commented on hadoop-ozone/dev-support/checks/acceptance.sh:
```diff
@@ -15,5 +15,6 @@
 DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
 export HADOOP_VERSION=3
-"$DIR/../../../hadoop-ozone/dist/target/ozone-*-SNAPSHOT/compose/test-all.sh"
+OZONE_VERSION=$(cat $DIR/../../pom.xml  | grep "<ozone.version>" | sed 's/<[^>]*>//g'|  sed 's/^[ \t]*//')
```
shellcheck:21: note: Double quote to prevent globbing and word splitting. [SC2086]
shellcheck:21: note: Useless cat. Consider 'cmd < file | ..' or 'cmd file | ..' instead. [SC2002]

hadoop-yetus commented on hadoop-ozone/dev-support/checks/checkstyle.sh:
```diff
@@ -13,7 +13,10 @@
+mvn -B -fn checkstyle:check -f pom.ozone.xml
+
+#Print out the exact violations with parsing XML results with sed
+find -name checkstyle-errors.xml | xargs sed  '$!N; /<file.*\n<\/file/d;P;D' | sed '/<\/.*/d;/<checkstyle.*/d;s/<error.*line="\([[:digit:]]*\)".*message="\([^"]\+\).*/ \1: \2/;s/<file name="\([^"]*\)".*/\1/;/<\?xml.*>/d'
```
shellcheck:1: note: Some finds don't have a default path. Specify '.' explicitly. [SC2185]

hadoop-yetus commented on hadoop-ozone/dev-support/checks/integration.sh:
```diff
@@ -0,0 +1,25 @@
+export MAVEN_OPTS="-Xmx4096m"
+mvn -B install -f pom.ozone.xml -DskipTests
+mvn -B -fn test -f pom.ozone.xml -pl :hadoop-ozone-integration-test,:hadoop-ozone-filesystem
+module_failed_tests=$(find "." -name 'TEST*.xml'\
```
shellcheck:23: warning: Use -print0/-0 or -exec + to allow for non-alphanumeric filenames. [SC2038]
shellcheck:49: note: This word is outside of quotes. Did you intend to 'nest '"'single quotes'"' instead'?  [SC2026]

hadoop-yetus commented on hadoop-ozone/dev-support/checks/rat.sh:
```diff
@@ -16,7 +16,10 @@
 mkdir -p target
 rm target/rat-aggregated.txt
-mvn -fn org.apache.rat:apache-rat-plugin:0.13:check -am -pl :hadoop-ozone-dist -Phdds
+cd hadoop-hdds
```
shellcheck:1: warning: Use 'cd ... || exit' or 'cd ... || return' in case cd fails. [SC2164]

hadoop-yetus commented on hadoop-ozone/dev-support/checks/rat.sh:
```diff
@@ -16,7 +16,10 @@
-mvn -fn org.apache.rat:apache-rat-plugin:0.13:check -am -pl :hadoop-ozone-dist -Phdds
+cd hadoop-hdds
+mvn -B -fn org.apache.rat:apache-rat-plugin:0.13:check
+cd ../hadoop-ozone
```
shellcheck:1: warning: Use 'cd ... || exit' or 'cd ... || return' in case cd fails. [SC2164]

hadoop-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 31 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| 0 | shelldocs | 1 | Shelldocs was not available. |
| 0 | @author | 0 | Skipping @author checks as author.sh has been patched. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 31 | Maven dependency ordering for branch |
| +1 | mvninstall | 472 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| -1 | pylint | 1 | Error running pylint. Please check pylint stderr files. |
| +1 | shadedclient | 726 | branch has no errors when building and testing our client artifacts. |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 31 | Maven dependency ordering for patch |
| +1 | mvninstall | 449 | the patch passed |
| +1 | mvnsite | 0 | the patch passed |
| -1 | pylint | 1 | Error running pylint. Please check pylint stderr files. |
| +1 | pylint | 1 | There were no new pylint issues. |
| -1 | shellcheck | 1 | The patch generated 8 new + 6 unchanged - 0 fixed = 14 total (was 6) |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 715 | patch has no errors when building and testing our client artifacts. |
||| _ Other Tests _ |
| +1 | unit | 88 | hadoop-hdds in the patch passed. |
| +1 | unit | 168 | hadoop-ozone in the patch passed. |
| +1 | asflicense | 36 | The patch does not generate ASF License warnings. |
| | | 2947 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1035/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1035 |
| Optional Tests | dupname asflicense mvnsite unit shellcheck shelldocs pylint |
| uname | Linux c87588353a8c 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / d203045 |
| pylint | https://builds.apache.org/job/hadoop-multibranch/job/PR-1035/1/artifact/out/branch-pylint-stderr.txt |
| pylint | https://builds.apache.org/job/hadoop-multibranch/job/PR-1035/1/artifact/out/patch-pylint-stderr.txt |
| shellcheck | https://builds.apache.org/job/hadoop-multibranch/job/PR-1035/1/artifact/out/diff-patch-shellcheck.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1035/1/testReport/ |
| Max. process+thread count | 413 (vs. ulimit of 5500) |
| modules | C: hadoop-ozone hadoop-ozone/fault-injection-test/network-tests U: hadoop-ozone |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1035/1/console |
| versions | git=2.7.4 maven=3.3.9 shellcheck=0.4.6 pylint=1.9.2 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

hadoop-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 32 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| 0 | shelldocs | 0 | Shelldocs was not available. |
| 0 | @author | 0 | Skipping @author checks as author.sh has been patched. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 52 | Maven dependency ordering for branch |
| +1 | mvninstall | 482 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| -1 | pylint | 5 | Error running pylint. Please check pylint stderr files. |
| +1 | shadedclient | 783 | branch has no errors when building and testing our client artifacts. |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 22 | Maven dependency ordering for patch |
| +1 | mvninstall | 431 | the patch passed |
| +1 | mvnsite | 0 | the patch passed |
| -1 | pylint | 6 | Error running pylint. Please check pylint stderr files. |
| +1 | pylint | 6 | There were no new pylint issues. |
| +1 | shellcheck | 0 | The patch generated 0 new + 0 unchanged - 7 fixed = 0 total (was 7) |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 703 | patch has no errors when building and testing our client artifacts. |
||| _ Other Tests _ |
| +1 | unit | 104 | hadoop-hdds in the patch passed. |
| +1 | unit | 176 | hadoop-ozone in the patch passed. |
| +1 | asflicense | 48 | The patch does not generate ASF License warnings. |
| | | 3007 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1035/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1035 |
| Optional Tests | dupname asflicense mvnsite unit shellcheck shelldocs pylint |
| uname | Linux 2c4cf55cdab1 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 1c254a8 |
| pylint | https://builds.apache.org/job/hadoop-multibranch/job/PR-1035/2/artifact/out/branch-pylint-stderr.txt |
| pylint | https://builds.apache.org/job/hadoop-multibranch/job/PR-1035/2/artifact/out/patch-pylint-stderr.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1035/2/testReport/ |
| Max. process+thread count | 444 (vs. ulimit of 5500) |
| modules | C: hadoop-ozone hadoop-ozone/fault-injection-test/network-tests U: hadoop-ozone |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1035/2/console |
| versions | git=2.7.4 maven=3.3.9 shellcheck=0.4.6 pylint=1.9.2 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

hadoop-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 32 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 1 | No case conflicting files found. |
| 0 | shelldocs | 1 | Shelldocs was not available. |
| 0 | @author | 0 | Skipping @author checks as author.sh has been patched. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 44 | Maven dependency ordering for branch |
| +1 | mvninstall | 504 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| -1 | pylint | 1 | Error running pylint. Please check pylint stderr files. |
| +1 | shadedclient | 772 | branch has no errors when building and testing our client artifacts. |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 32 | Maven dependency ordering for patch |
| +1 | mvninstall | 454 | the patch passed |
| +1 | mvnsite | 0 | the patch passed |
| -1 | pylint | 2 | Error running pylint. Please check pylint stderr files. |
| +1 | pylint | 2 | There were no new pylint issues. |
| +1 | shellcheck | 1 | The patch generated 0 new + 0 unchanged - 7 fixed = 0 total (was 7) |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 708 | patch has no errors when building and testing our client artifacts. |
||| _ Other Tests _ |
| +1 | unit | 103 | hadoop-hdds in the patch passed. |
| +1 | unit | 179 | hadoop-ozone in the patch passed. |
| +1 | asflicense | 48 | The patch does not generate ASF License warnings. |
| | | 3076 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1035/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1035 |
| Optional Tests | dupname asflicense mvnsite unit shellcheck shelldocs pylint |
| uname | Linux f16dcd073b52 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 9c90729 |
| pylint | https://builds.apache.org/job/hadoop-multibranch/job/PR-1035/3/artifact/out/branch-pylint-stderr.txt |
| pylint | https://builds.apache.org/job/hadoop-multibranch/job/PR-1035/3/artifact/out/patch-pylint-stderr.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1035/3/testReport/ |
| Max. process+thread count | 446 (vs. ulimit of 5500) |
| modules | C: hadoop-ozone hadoop-ozone/fault-injection-test/network-tests U: hadoop-ozone |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1035/3/console |
| versions | git=2.7.4 maven=3.3.9 shellcheck=0.4.6 pylint=1.9.2 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

hadoop-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 32 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| 0 | shelldocs | 0 | Shelldocs was not available. |
| 0 | @author | 0 | Skipping @author checks as author.sh has been patched. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 32 | Maven dependency ordering for branch |
| +1 | mvninstall | 474 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| -1 | pylint | 1 | Error running pylint. Please check pylint stderr files. |
| +1 | shadedclient | 774 | branch has no errors when building and testing our client artifacts. |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 30 | Maven dependency ordering for patch |
| +1 | mvninstall | 453 | the patch passed |
| +1 | mvnsite | 0 | the patch passed |
| -1 | pylint | 2 | Error running pylint. Please check pylint stderr files. |
| +1 | pylint | 2 | There were no new pylint issues. |
| +1 | shellcheck | 2 | The patch generated 0 new + 0 unchanged - 7 fixed = 0 total (was 7) |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 702 | patch has no errors when building and testing our client artifacts. |
||| _ Other Tests _ |
| +1 | unit | 104 | hadoop-hdds in the patch passed. |
| +1 | unit | 179 | hadoop-ozone in the patch passed. |
| +1 | asflicense | 49 | The patch does not generate ASF License warnings. |
| | | 3029 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1035/4/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1035 |
| Optional Tests | dupname asflicense mvnsite unit shellcheck shelldocs pylint |
| uname | Linux cc2fdb9998fb 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 9c90729 |
| pylint | https://builds.apache.org/job/hadoop-multibranch/job/PR-1035/4/artifact/out/branch-pylint-stderr.txt |
| pylint | https://builds.apache.org/job/hadoop-multibranch/job/PR-1035/4/artifact/out/patch-pylint-stderr.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1035/4/testReport/ |
| Max. process+thread count | 411 (vs. ulimit of 5500) |
| modules | C: hadoop-ozone hadoop-ozone/fault-injection-test/network-tests U: hadoop-ozone |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1035/4/console |
| versions | git=2.7.4 maven=3.3.9 shellcheck=0.4.6 pylint=1.9.2 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

anuengineer commented:
@elek  I was trying to merge, but seems like we have some conflicts. Perhaps due to the fact that I merged a patch from nanda, also there is an author check warning.

hadoop-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 69 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 1 | No case conflicting files found. |
| 0 | shelldocs | 1 | Shelldocs was not available. |
| 0 | @author | 0 | Skipping @author checks as author.sh has been patched. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 50 | Maven dependency ordering for branch |
| +1 | mvninstall | 532 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| -1 | pylint | 8 | Error running pylint. Please check pylint stderr files. |
| +1 | shadedclient | 792 | branch has no errors when building and testing our client artifacts. |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 32 | Maven dependency ordering for patch |
| +1 | mvninstall | 474 | the patch passed |
| +1 | mvnsite | 0 | the patch passed |
| -1 | pylint | 8 | Error running pylint. Please check pylint stderr files. |
| +1 | pylint | 8 | There were no new pylint issues. |
| +1 | shellcheck | 0 | The patch generated 0 new + 0 unchanged - 7 fixed = 0 total (was 7) |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 669 | patch has no errors when building and testing our client artifacts. |
||| _ Other Tests _ |
| +1 | unit | 99 | hadoop-hdds in the patch passed. |
| +1 | unit | 175 | hadoop-ozone in the patch passed. |
| +1 | asflicense | 40 | The patch does not generate ASF License warnings. |
| | | 3152 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=18.09.7 Server=18.09.7 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1035/5/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1035 |
| Optional Tests | dupname asflicense mvnsite unit shellcheck shelldocs pylint |
| uname | Linux 3e3d75757cf8 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 9119ed0 |
| pylint | https://builds.apache.org/job/hadoop-multibranch/job/PR-1035/5/artifact/out/branch-pylint-stderr.txt |
| pylint | https://builds.apache.org/job/hadoop-multibranch/job/PR-1035/5/artifact/out/patch-pylint-stderr.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1035/5/testReport/ |
| Max. process+thread count | 423 (vs. ulimit of 5500) |
| modules | C: hadoop-ozone hadoop-ozone/fault-injection-test/network-tests U: hadoop-ozone |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1035/5/console |
| versions | git=2.7.4 maven=3.3.9 shellcheck=0.4.6 pylint=1.9.2 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

hadoop-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 38 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| 0 | shelldocs | 0 | Shelldocs was not available. |
| 0 | @author | 0 | Skipping @author checks as author.sh has been patched. |
| -1 | test4tests | 0 | The patch doesn't appear to include any new or modified tests.  Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch. |
||| _ trunk Compile Tests _ |
| +1 | mvninstall | 469 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 732 | branch has no errors when building and testing our client artifacts. |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 426 | the patch passed |
| +1 | mvnsite | 0 | the patch passed |
| +1 | shellcheck | 1 | The patch generated 0 new + 0 unchanged - 7 fixed = 0 total (was 7) |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 632 | patch has no errors when building and testing our client artifacts. |
||| _ Other Tests _ |
| +1 | unit | 98 | hadoop-hdds in the patch passed. |
| +1 | unit | 176 | hadoop-ozone in the patch passed. |
| +1 | asflicense | 41 | The patch does not generate ASF License warnings. |
| | | 2788 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=18.09.7 Server=18.09.7 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1035/6/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1035 |
| Optional Tests | dupname asflicense mvnsite unit shellcheck shelldocs |
| uname | Linux 36fd912879ea 4.4.0-139-generic #165-Ubuntu SMP Wed Oct 24 10:58:50 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 9119ed0 |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1035/6/testReport/ |
| Max. process+thread count | 446 (vs. ulimit of 5500) |
| modules | C: hadoop-ozone U: hadoop-ozone |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1035/6/console |
| versions | git=2.7.4 maven=3.3.9 shellcheck=0.4.6 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

elek commented:
> @elek I was trying to merge, but seems like we have some conflicts. Perhaps due to the fact that I merged a patch from nanda, also there is an author check warning.

Yes, I rebased in on the top of the trunk. 

Author check is false positive as one of the scripts greps for "@author" tags, obviously it contains the @author inside as string. I changed it to use string concatenation (@a + uthor) to make yetus happy....
